satellite/satellitedb/identdb.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/storj"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type certDB struct {
+	db *dbx.DB
+}
+
+// Set adds a peer identity entry
+func (certs *certDB) Set(ctx context.Context, nodeID storj.NodeID, peerIdent *identity.PeerIdentity) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	tx, err := certs.db.Begin()
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+		} else {
+			err = errs.Combine(err, tx.Rollback())
+		}
+	}()
+
+	if peerIdent == nil {
+		return Error.New("Peer Identity cannot be nil")
+	}
+	chain := identity.EncodePeerIdentity(peerIdent)
+
+	var id []byte
+	query := `SELECT node_id FROM peer_identities WHERE serial_number = ?;`
+	err = tx.QueryRow(certs.db.Rebind(query), peerIdent.Leaf.SerialNumber.Bytes()).Scan(&id)
```

Comment:
You need to select on both the `node_id` and `serial_number`.

Reply:
That might create a problem, as lets say there exists an entry for a nodeID and serial#, not a request to update the same node id, and with a different serial#. The query with latest nodeID and new serial# would return no sql rows, and when I try to `INSERT` it will cause an error as nodeID is set as `UNIQUE`. The query has to be made with nodeID and get serial number. Compare the serial number with the serial number you try to update. If they match just return, otherwise update.